internal/pkg/policy/policy_output.go

```diff
@@ -97,6 +111,64 @@ func (p *Output) prepareElasticsearch(
 		agent.Outputs[p.Name] = output
 	}
 
+	// retire api key of removed remote output
+	var toRetireAPIKeys *model.ToRetireAPIKeyIdsItems
+	var removedOutputKey string
+	// find the first output that is removed - supposing one output can be removed at a time
+	for agentOutputKey, agentOutput := range agent.Outputs {
+		found := false
+		for outputMapKey := range outputMap {
+			if agentOutputKey == outputMapKey {
+				found = true
+			}
+		}
+		if !found {
+			zlog.Info().Str("APIKeyID", agentOutput.APIKeyID).Str("output", agentOutputKey).Msg("Output removed, will retire API key")
```

Comment:
use the [ecs compliant](https://github.com/elastic/fleet-server/blob/main/internal/pkg/logger/ecs.go#L63) label?

Reply:
⚠️  Does `agentOutputKey` contain the credential? If so we cannot log this out. If not, can we rename this to be less scary looking on a log line?

Reply:
No, it is only the output name. I'll rename.

Reply:
done